src/java/org/apache/poi/ss/usermodel/DataFormatter.java

```diff
@@ -691,6 +677,57 @@ private String cleanFormatForNumber(String formatStr) {
         return sb.toString();
     }
 
+    private static class InternalDecimalFormatWithScale extends Format {
+
+        private static final Pattern endsWithCommas = Pattern.compile("(,+)$");
+        private BigDecimal divider;
+        private static final BigDecimal ONE_THOUSAND = new BigDecimal(1000);
+        private final DecimalFormat df;
+        private static final String trimTrailingCommas(String s) {
+            return s.replaceAll(",+$", "");
+        }
+
+        public InternalDecimalFormatWithScale(String pattern, DecimalFormatSymbols symbols) {
+            df = new DecimalFormat(trimTrailingCommas(pattern), symbols);
+            setExcelStyleRoundingMode(df);
+            Matcher endsWithCommasMatcher = endsWithCommas.matcher(pattern);
+            if (endsWithCommasMatcher.find()) {
+                String commas = (endsWithCommasMatcher.group(1));
+                BigDecimal temp = BigDecimal.ONE;
```

Comment:
Just another quick note. The implementation is based on the following comment in the ECMA-376 5th edition specs:

>**,** Thousands separator. The application shall separate thousands by commas if the format
contains a comma that is enclosed by number signs (#) or by zeros. A comma that follows
a placeholder scales the number by one thousand. [Example: If the format is #.0,, and the
cell value is 12,200,000 then the number 12.2 is displayed. end example]

see section 18.8.31 numFmts (Number Formats)